ce/ce/cli/commands/find.ts

```diff
@@ -42,14 +43,30 @@ export class FindCommand extends Command {
     const table = new Table('Artifact', 'Version', 'Summary');
 
     for (const each of this.inputs) {
-      for (const [registry, id, artifacts] of await registries.search({ keyword: each, version: this.version.value })) {
-        const latest = artifacts[0];
-        if (!latest.metadata.info.dependencyOnly) {
-          const name = artifactIdentity(latest.registryId, id, latest.shortName);
-          table.push(name, latest.metadata.info.version, latest.metadata.info.summary || '');
+      // eslint-disable-next-line prefer-const
+      for (let [registry, id, artifacts] of await registries.search({
+        // use keyword search if no registry is specified
+        keyword: each.indexOf(':') === -1 ? each : undefined,
+        // otherwise use the criteria as an id
+        idOrShortName: each.indexOf(':') > -1 ? each : undefined,
```

Comment:
1. Shouldn't idOrShortName be the part after the : rather than the whole thing?
2. Could you introduce `const registrySpecified = each.indexOf(':') !== -1;` which would avoid the need for some of the comments?

Reply:
The actual search uses the full value with the colon to scope the search